sympy/functions/combinatorial/numbers.py

```diff
@@ -1861,9 +1861,29 @@ def stirling(n, k, d=None, kind=2, signed=False):
 def _nT(n, k):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
+    if k > n or k < 0:
+        return 0
+    if k == n or k == 1:
+        return 1
     if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+        return 0
+    if k == 2:
+        return n//2
+    d = n - k
+    if d <= 2:
```

Comment:
I believe the 2 can be changed to 3 with the given conditions preceding it.

Reply:
@smichr  I changed it.